SMT-LIB 2 parser: implement unary minus

kroening opened the pull request:
<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.
--->

- [X] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

kroening committed:
SMT-LIB 2 parser: implement unary minus

hannes-steffenhagen-diffblue reviewed (commented):
Change looks OK to me, but this isn't covered by a test?

kroening committed:
smt2_solver: fix simplify command

The simplify command now prints the simplified expression, as opposed to the
argument given.

kroening committed:
test for simplify on integers

This test covers parsing unary minus on integers, the simplify command, and
printing negative integer literals.

kroening committed:
smt2: fix negative integer literals

Integer literals that are negative need to be printed as (- x), as opposed
to -x, which parses as a symbol.
